### ISSUE TITLE
[Bug]: mobile sync error

### ISSUE THREAD

**Issue by imeed166** [remotely-save/remotely-save#55]
### What happened?

It works fine on Mac, but when I use `start sync` command on android, I get this 2 notification errors after `6/8 Starting to generate sync plan.`
```
remotely-save-1647674854846: abort sync, triggerSource=manual, error while generating_plan
```
```
Error: File ###.md has a last modified time >=0 or undefined in the local file system. It's abnormal and the plugin stops.
```
I haven't modified any file on my phone. I just copied the entire vault and started syncing.

### What OS are you using?

Android

### What remote cloud services are you using?

Dropbox

### Version of the plugin

0.3.7

### Version of Obsidian

mobile v.1.1.0

### Ensure no sensitive information

- [X] I ensure that no sensitive information is submitted in the issue.

**Comment by imeed166** [comment]
When I delete the file responsible, the same error repeats with another file.

**Comment by fyears** [comment]
It basically means that the “last modified time” or “creation time” of files on your phone are abnormal, returning 0, or equivalently the date 1970-01-01. You could use any file explorer app to verify the times of your files.

Unfortunately, the plugin can do nothing in this case, because it heavily relies on the times to know which “version” is newer locally and remotel. I suggest you fine out the reason why the times of the files are abnormal.

**Comment by imeed166** [comment]
I copied a compressed version of the vault, then extracted it before opening the vault with obsidian mobile. Maybe that's what's causing the weird creation times. 

Thank you for the clarification.

**Comment by imeed166** [comment]
I checked one of the files that's causing this, and the last modified time is `Sep 18, 2021 7:40pm` is this time supposed to give an error ?

**Comment by imeed166** [comment]
I'm noticing 2 modified time formats :
- `MMM DD, YYYY HH:MM A` for .jpg files, like `Sep 18, 2021 11:50 AM`
- `MMM DD HH:mm A` for .md files, like `Sep 18 11:50 AM`

Waybe that's what's causing the error.

**Comment by imeed166** [comment]
Sorry to comment on a closed issue, but It's still happening even though I synced everything through remotely save.

**Steps to reproduce :**
1. clean vault on obsidian mobile
2. install remotely save
3. start syncing, everything works fine
4. In the second sync, the same error happens again

**Comment by fyears** [comment]
That’s ok, but is there any abnormal last modified time again?

**Comment by fyears** [comment]
would you please follow this, and checkout is ther any 0 mtimeLocal files? If so , then that’s still the problem.

https://github.com/remotely-save/remotely-save/blob/master/docs/how_to_debug/export_sync_plans.md

**Comment by imeed166** [comment]
> would you please follow this, and checkout is there any 0 mtimeLocal files? If so , then that’s still the problem.

Okay, I searched the `sync_plan_hist...` for `mtimeLocal` and didn't find anything.

(Btw, I switched to using one drive because I couldn't authorize dropbox on mobile)

> is there any abnormal last modified time again?

When I checked the files in the file explorer, All the formats were the same.

**Comment by fyears** [comment]
I am quite sure that the error comes here it does nothing but checking the file mtime…

https://github.com/remotely-save/remotely-save/blob/63a66d69f9a9fb98138adfe46dea2d5da0475a23/src/sync.ts#L453

**Comment by imeed166** [comment]
> And no any dates from 1970 in file explorer?

nothing like that

> So, no any 0 mtimelocal and mtimeremote in sync_plan_hist... files?

nothing like that

Here's one mtimeremote for a file that triggers the error `1647846480660`

**Comment by imeed166** [comment]
When I compare this file with the others in the `sync_plan_hist...`, It was the first with `decisionBranch: 5` the ones before were `decisionBranch: 13 or 11`

**Comment by fyears** [comment]
Hi, I just released a new version [0.3.11](https://github.com/remotely-save/remotely-save/releases/tag/0.3.11) showing more error information to the screen.

1. Please install the new version
2. Please go to the Remotely Save settings page, scroll down to very last of the page, you will see "reset local internal cache/databases" (in the section "Debug"), then click the button "Reset". 
3. Then **restart the Obsidian**.
4. And start the sync again.
5. Pay attention to the (new) error.

You shall see something like the screenshot, 
and please remember that, the "key" and "remoteEncryptedKey" are usually **NOT** supposed to be shown to anyone, because that's your file name!

* Expected behavior: `"existLocal": true, "mtimeLocal": a number > 0`
* One unexpected behavior: `"existLocal": true, "mtimeLocal": 0`
* Another unexpected behavior: `"existLocal": true`, and `"mtimeLocal"` field does NOT exist

![img](https://user-images.githubusercontent.com/1142836/159323992-a6cc47ac-b9b3-41dd-9090-8b1faf0a8afc.png)

**Comment by fyears** [comment]
> When I compare this file with the others in the `sync_plan_hist...`, It was the first with `decisionBranch: 5`, the ones before were `decisionBranch: 13 or 11`

And, the error (`a last modified time <=0 or undefined in the local file system`) should show up **BEFORE** generating any `decisionBranch`.

**Comment by imeed166** [comment]
![20220321_182423](https://user-images.githubusercontent.com/38974541/159329738-2a6cfd7f-fd3c-4082-9d10-d7f72631ac70.jpg)

**Comment by imeed166** [comment]
I think, It has to do with a bug in the mobile app 
![Screen Shot 2022-03-21 at 6 48 40 PM](https://user-images.githubusercontent.com/38974541/159333595-ce2e3794-5a84-49ac-85f5-b0bfeebf18fc.jpg)

I'll wait for the next update and tell you if it fixes the problem.

**Comment by imeed166** [comment]
Yep, after the update, everything works. Thank you so much for your time and efforts.

**Comment by fyears** [comment]
Wow, 1.1.1 is just a new update today! So I believe all you need is updating your mobile Obsidian, haha!
